packages/state-transition/src/util/epochShuffling.ts

```diff
@@ -109,9 +110,46 @@ export function computeEpochShuffling(
 ): EpochShuffling {
   const activeValidatorCount = activeIndices.length;
 
-  const shuffling = activeIndices.slice();
   const seed = getSeed(state, epoch, DOMAIN_BEACON_ATTESTER);
-  unshuffleList(shuffling, seed);
+  const shuffling = unshuffleList(activeIndices, seed, SHUFFLE_ROUND_COUNT);
+
+  const committeesPerSlot = computeCommitteeCount(activeValidatorCount);
+
+  const committeeCount = committeesPerSlot * SLOTS_PER_EPOCH;
+
+  const committees: Uint32Array[][] = [];
+  for (let slot = 0; slot < SLOTS_PER_EPOCH; slot++) {
+    const slotCommittees: Uint32Array[] = [];
+    for (let committeeIndex = 0; committeeIndex < committeesPerSlot; committeeIndex++) {
+      const index = slot * committeesPerSlot + committeeIndex;
+      const startOffset = Math.floor((activeValidatorCount * index) / committeeCount);
+      const endOffset = Math.floor((activeValidatorCount * (index + 1)) / committeeCount);
+      if (!(startOffset <= endOffset)) {
+        throw new Error(`Invalid offsets: start ${startOffset} must be less than or equal end ${endOffset}`);
+      }
+      slotCommittees.push(shuffling.subarray(startOffset, endOffset));
+    }
+    committees.push(slotCommittees);
+  }
+
+  return {
+    epoch,
+    activeIndices,
+    shuffling,
+    committees,
+    committeesPerSlot,
+  };
+}
+
+export async function computeEpochShufflingAsync(
+  state: BeaconStateAllForks,
+  activeIndices: Uint32Array,
+  epoch: Epoch
+): Promise<EpochShuffling> {
+  const activeValidatorCount = activeIndices.length;
+
+  const seed = getSeed(state, epoch, DOMAIN_BEACON_ATTESTER);
+  const shuffling = await asyncUnshuffleList(activeIndices, seed, SHUFFLE_ROUND_COUNT);
```

Comment:
can `@chainsafe/swap-or-not-shuffle` provide a sync version? need a benchmark to see how much faster it is compared to NodeJS version

Reply:
Perf tests were added to `@chainsafe/swap-or-not-shuffle` repo to compare the two as well as the async implementation.

```sh
  shuffle list - 16384 indices
    ✓ JS - unshuffleList - 16384 indices                                  2815.593 ops/s    355.1650 us/op        -        852 runs  0.690 s
    ✓ Rust - unshuffleList - 16384 indices                                3797.084 ops/s    263.3600 us/op        -       3450 runs   1.17 s
    ✓ Rust - asyncUnshuffleList - 16384 indices                           3470.800 ops/s    288.1180 us/op        -       3184 runs   1.22 s

  shuffle list - 250000 indices
    ✓ JS - unshuffleList - 250000 indices                                 140.1907 ops/s    7.133141 ms/op        -         46 runs  0.829 s
    ✓ Rust - unshuffleList - 250000 indices                               231.9156 ops/s    4.311914 ms/op        -        143 runs   1.12 s
    ✓ Rust - asyncUnshuffleList - 250000 indices                          233.1750 ops/s    4.288624 ms/op        -         73 runs  0.816 s

  shuffle list - 1000000 indices
    ✓ JS - unshuffleList - 1000000 indices                                33.88110 ops/s    29.51498 ms/op        -         17 runs   1.02 s
    ✓ Rust - unshuffleList - 1000000 indices                              56.83460 ops/s    17.59492 ms/op        -         31 runs   1.05 s
    ✓ Rust - asyncUnshuffleList - 1000000 indices                         57.15035 ops/s    17.49771 ms/op        -         25 runs  0.944 s
```